pages/home.py

```diff
@@ -413,6 +506,260 @@ def functional_narrative(df):
     return text
 
 
+def subnational_spending_narrative(
+    df_spending,
+    df_poverty,
+    top_n=3,
+    exp_thresh=0.5,
+    per_capita_thresh=1000,
+    corr_thresholds=(0.3, 0.7),
+):
+    total_expenditure = (
+        df_spending.groupby("adm1_name")["expenditure"]
+        .sum()
+        .sort_values(ascending=False)
+    )
+    top_n_total = total_expenditure.head(top_n)
+    total_expenditure_sum = total_expenditure.sum()
+    top_n_percentage = top_n_total.sum() / total_expenditure_sum
+    per_capita_expenditure = df_spending.groupby("adm1_name")[
+        "per_capita_expenditure"
+    ].mean()
+    per_capita_range = per_capita_expenditure.max() - per_capita_expenditure.min()
+    per_capita_median = per_capita_expenditure.median()
+    if not df_poverty.empty:
+        poverty_rates = df_poverty.groupby("region_name")["poor215"].mean()
+        correlation = per_capita_expenditure.corr(poverty_rates)
+
+    if top_n_percentage > exp_thresh:
+        exp_narrative = f"The top {top_n} regions—{', '.join(top_n_total.index)}—account for {top_n_percentage:.1%} of the total government expenditure,\
+            indicating a significant concentration in these areas."
+    else:
+        exp_narrative = f"The top {top_n} regions—{', '.join(top_n_total.index)}—account for {top_n_percentage:.1%} of the total government expenditure."
+
+    if per_capita_range > per_capita_thresh:
+        per_capita_narrative = f"Per capita spending varies widely across regions, ranging from {per_capita_expenditure.min():,.2f} \
+            to {per_capita_expenditure.max():,.2f}, with a median of {per_capita_median:,.2f}. This indicates substantial variation in resource allocation per person."
+    else:
+        per_capita_narrative = f"Per capita spending ranges from {per_capita_expenditure.min():,.2f} to {per_capita_expenditure.max():,.2f}, \
+            with a median of {per_capita_median:,.2f}. The distribution is relatively even across regions."
+    if not df_poverty.empty:
+        if abs(correlation) > corr_thresholds[1]:
+            corr_narrative = f"The correlation between per capita spending and poverty rates is {correlation:.2f},\
+                indicating a strong inverse relationship. Higher per capita spending is generally associated with lower poverty rates."
+        elif abs(correlation) > corr_thresholds[0]:
+            corr_narrative = f"The correlation between per capita spending and poverty rates is {correlation:.2f} (Pearson correlation coefficient), \
+                suggesting a moderate inverse relationship. Generally, higher per capita spending is associated with lower poverty, though exceptions exist."
```

Comment:
Let's remove all the references to (Pearson correlation coefficient)

Reply:
Oh. Sorry. Not sure how I missed that. Made the changes